packages/admin/cms-admin/src/pages/createEditPageNode.tsx

```diff
@@ -30,24 +31,85 @@ import {
 import { useLocale } from "../locale/useLocale";
 import { automaticRedirectsRefetchQueryDescription } from "../redirects/RedirectsTable.gql";
 
-const editPageNodeQuery = gql`
-    query EditPageNode($id: ID!) {
-        page: pageTreeNode(id: $id) {
-            id
-            name
-            slug
-            path
-            documentType
-            hideInMenu
-            parentId
-            document {
-                ... on DocumentInterface {
-                    id
+type SerializedInitialValues = string;
+
+export interface EditPageNodeFinalFormValues {
+    [key: string]: unknown;
+}
+
+interface CreateEditPageNodeProps {
+    additionalFormFields?: React.ReactNode;
+    nodeFragment?: { name: string; fragment: DocumentNode };
+    valuesToInput?: (values: EditPageNodeFinalFormValues) => EditPageNodeFinalFormValues;
+}
+
+export interface EditPageNodeProps {
+    id: null | SerializedInitialValues | string; // when mode is add: SerializedInitialValues is expected
+    mode: "add" | "edit";
+    category: GQLPageTreeNodeCategory;
+    documentTypes: Record<DocumentType, DocumentInterface>;
+}
+
+export function createEditPageNode({
+    additionalFormFields,
+    nodeFragment,
+    valuesToInput,
+}: CreateEditPageNodeProps): (props: EditPageNodeProps) => JSX.Element {
+    const editPageNodeQuery = gql`
+        query EditPageNode($id: ID!) {
+            page: pageTreeNode(id: $id) {
+                id
+                name
+                slug
+                path
+                documentType
+                hideInMenu
+                parentId
+                document {
+                    ... on DocumentInterface {
+                        id
+                    }
                 }
+                ${nodeFragment ? "...".concat(nodeFragment?.name) : ""}
             }
         }
-    }
-`; // @TODO: use a graphql interface to ommit the ...on (DocumentInterface)
+        ${nodeFragment ? nodeFragment?.fragment : ""}
+    `; // @TODO: use a graphql interface to ommit the ...on (DocumentInterface)
+
+    const updatePageNodeMutation = gql`
+        mutation UpdatePageNode($nodeId: ID!, $input: PageTreeNodeUpdateInput!) {
+            updatePageTreeNode(id: $nodeId, input: $input) {
+                id
+                name
+                slug
+                documentType
+                hideInMenu
+                document {
+                    ... on DocumentInterface {
+                        id
+                    }
+                }
+                ${nodeFragment ? "...".concat(nodeFragment?.name) : ""}
+            }
+        }
+        ${nodeFragment ? nodeFragment?.fragment : ""}
+    `;
+
+    return function useEditPageNode({ id, mode, documentTypes, category }: EditPageNodeProps): JSX.Element {
```

Comment:
The factory should return the `EditPageNode` function component directly. Props such as `editPageNodeQuery` should not be necessary.

Reply:
done